multiqc/modules/vep/vep.py

```diff
@@ -0,0 +1,365 @@
+                if len(cells) == 2:
+                    key = cells[0][4:-5]
+                    value = cells[1][4:-5]
+                    if key == "Novel / existing variants":
+                        values = value.split("/")
+                        novel = values[0].split("(")[0].replace(" ", "")
+                        existing = values[1].split("(")[0].replace(" ", "")
+                        self.vep_data[f["s_name"]][title]["Novel variants"] = int(novel)
+                        self.vep_data[f["s_name"]][title]["Existing variants"] = int(existing)
+                    else:
+                        self.vep_data[f["s_name"]][title][key] = int(value)
+
+    def parse_vep_txt(self, f):
+        """This Function will parse VEP summary files with plain text format"""
+        # Initialise vep_data dictionary for the current sample
+        if f["s_name"] not in self.vep_data:
+            self.vep_data[f["s_name"]] = {}
+        vepf = f["f"]
+        txt_data = {}
+        title = ""
+        for line in vepf.readlines():
+            line = line.rstrip("\n")
+            if len(line) < 1:
+                continue
+            if line[0] == "[":
+                title = line.replace("[", "").replace("]", "")
+                txt_data[title] = {}
+                continue
+            key, value = line.split("\t")
+            if key == "Novel / existing variants":
+                values = value.split("/")
+                novel = values[0].split("(")[0].replace(" ", "")
+                existing = values[1].split("(")[0].replace(" ", "")
+                txt_data[title]["Novel variants"] = int(novel)
+                txt_data[title]["Existing variants"] = int(existing)
+            elif title != "VEP run statistics":
+                txt_data[title][key] = int(value)
+            else:
+                txt_data[title][key] = value
+
+        for title in txt_data.keys():
+            if title not in self.vep_data[f["s_name"]]:
+                self.vep_data[f["s_name"]][title] = {}
+            self.vep_data[f["s_name"]][title] = txt_data[title]
+
+    def add_stats_table(self):
+        """Add a section with VEP General Statistics"""
+        table_config = {
+            "id": "vep-general-stats",
+            "namespace": "VEP",
+            "table_title": "VEP General Statistics",
+        }
+        table_data = {s_name: self.vep_data[s_name]["General statistics"] for s_name in self.vep_data}
+
+        # Build the header configs
+        cat_names = [
+            "Overlapped regulatory features",
+            "Overlapped transcripts",
+            "Overlapped genes",
+            "Existing variants",
+            "Novel variants",
+            "Variants filtered out",
+            "Variants processed",
+            "Lines of input read",
+        ]
+        # Set up the base config for each column
+        table_cats = OrderedDict()
+        color_list = ["Oranges", "Reds", "Blues", "Greens"]
```

Comment:
In the end after a bit of order change the colours ended up being ok so I dropped my whinging on this 😅 Main thing is to make it easy to scan / differentiate between columns and this does that 👍🏻